V0_16_UPGRADE_GUIDE.md

```diff
@@ -0,0 +1,20 @@
+# Version 0.16.0 Upgrade Guide
```

Comment:
I would put this in a `docs` folder, that's our canonical place to store documentation.

Reply:
Done.